MixItUp.Desktop/Services/DesktopSettingsService.cs

```diff
@@ -187,16 +185,35 @@ public string GetDatabaseFilePath(IChannelSettings settings)
             return Path.Combine(SettingsDirectoryName, string.Format("{0}.{1}.sqlite", settings.Channel.id.ToString(), (settings.IsStreamer) ? "Streamer" : "Moderator"));
         }
 
-        private async Task<IChannelSettings> LoadSettings(string filePath)
+        public async Task<int> GetSettingsVersion(string filePath)
         {
             string fileData = await ChannelSession.Services.FileService.ReadFile(filePath);
+            if (string.IsNullOrEmpty(fileData))
+            {
+                return -1;
+            }
+
             JObject settingsJObj = JObject.Parse(fileData);
-            int currentVersion = (int)settingsJObj["Version"];
+            return (int)settingsJObj["Version"];
+        }
 
+        public int GetLatestVersion()
+        {
+            return DesktopChannelSettings.LatestVersion;
+        }
+
+        private async Task<IChannelSettings> LoadSettings(string filePath)
+        {
+            int currentVersion = await GetSettingsVersion(filePath);
             if (currentVersion < DesktopChannelSettings.LatestVersion)
             {
                 await DesktopSettingsUpgrader.UpgradeSettingsToLatest(currentVersion, filePath);
             }
+            else if (currentVersion > DesktopChannelSettings.LatestVersion)
+            {
+                // Future build, like a preview build, we can't load this
+                return null;
+            }
 
```

Comment:
The above code will ensure we don't try to deserialize a newer settings file. This can happen if the user manually copies in the newer settings file.